src/Options.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace LanguageServer;
+
+class Options
+{
+    /**
+     * Filetypes the indexer should process
+     *
+     * @var array
+     */
+    private $fileTypes = [".php"];
+
+    /**
+     * @param \Traversable|\stdClass|array|null $options
+     */
+    public function __construct($options = null)
+    {
+        // Do nothing when the $options parameter is not an object
+        if (!is_object($options) && !is_array($options) && (!$options instanceof \Traversable)) {
+            return;
+        }
+
+        foreach ($options as $option => $value) {
+            $method = 'set' . ucfirst($option);
+
+            call_user_func([$this, $method], $value);
+        }
```

Comment:
I would rather let `JsonMapper` do this (like it does for all protocol structures) instead of doing this here